cmd/ion-go/eventwriter.go

```diff
@@ -15,6 +15,7 @@ type eventwriter struct {
 	depth       int
 	fieldname   string
 	annotations []string
+	inStruct    map[int]bool
```

Comment:
Could we re-use the `ctxstack` concept that's leveraged by the reader? That would avoid heap allocations.

Reply:
```ctxstack``` is internal right now and can be only used within ```ion``` directory.
Do we want to make ```ctxstack``` public?

Reply:
We can move ```ctx.go``` to 'internal' directory in order to be able use that in cmd package. But then we will have cycle imports (as ctx is using ion.Type).
A similar case is under discussion/investigation in an [ion-hash-go pull request](https://github.com/amzn/ion-hash-go/pull/31).

Reply:
To unlock Ion-hash, can we merge this pr and revisit this concern in #91 ?